demos/python_demos/cocosnet_demo/cocosnet_demo.py

```diff
@@ -0,0 +1,142 @@
+                           "Absolute MKLDNN (CPU)-targeted custom layers. Absolute path to a shared library with the "
+                           "kernels implementations", type=str, default=None)
+    args.add_argument("-d", "--device",
+                      help="Optional. Specify the target device to infer on; CPU, GPU, FPGA, HDDL or MYRIAD is "
+                           "acceptable. Sample will look for a suitable plugin for device specified. Default value is CPU",
+                      default="CPU", type=str)
+    return parser
+
+
+def get_files(path):
+    if os.path.isdir(path):
+        return os.listdir(path)
+    return [""]
+
+
+def get_mask_from_image(image, model):
+    image = preprocess_for_seg_model(image)
+    res = model.infer(image)
+    mask = np.argmax(res, axis=1)
+    mask = np.squeeze(mask, 0)
+    return mask + 1
+
+
+def main():
+    log.basicConfig(format="[ %(levelname)s ] %(message)s", level=log.INFO, stream=sys.stdout)
+    args = build_argparser().parse_args()
+
+    log.info("Creating CoCosNet Model")
+    ie_core = IECore()
+    if args.cpu_extension and 'CPU' in args.device:
+        ie_core.add_extension(args.cpu_extension, "CPU")
+    corr_model = CorrespondenceModel(ie_core, args.correspondence_model,
+                                     args.correspondence_model.replace(".xml", ".bin"),
+                                     args.device)
+    gen_model = GenerativeModel(ie_core, args.generative_model,
+                                args.generative_model.replace(".xml", ".bin"),
+                                args.device)
+    seg_model = SegmentationModel(ie_core, args.segmentation_model,
+                                  args.segmentation_model.replace(".xml", ".bin"),
+                                  args.device) if args.segmentation_model else None
+    model = CocosnetModel(corr_model, gen_model)
+
+    log.info("Preparing input data")
+    input_data = []
+    assert args.reference_images and ((args.input_semantics and
+           args.reference_semantics) or args.input_images), "Not enough data to do inference"
+    input_images = sorted(get_files(args.input_images))
+    input_semantics = sorted(get_files(args.input_semantics))
+    reference_images = sorted(get_files(args.reference_images))
+    reference_semantics = sorted(get_files(args.reference_semantics))
+    n = len(reference_images)
+    if seg_model:
+        players = [input_images, n * [''], reference_images, n * ['']]
+    else:
+        players = [n * [''], input_semantics, reference_images, reference_semantics]
+    for input_img, input_sem, ref_img, ref_sem in zip(*players):
+        if seg_model:
+            input_sem = get_mask_from_image(cv2.imread(args.input_images + input_img), seg_model)
+            ref_sem = get_mask_from_image(cv2.imread(args.reference_images + ref_img), seg_model)
+        else:
+            input_sem = cv2.imread(args.input_semantics + input_sem, cv2.IMREAD_GRAYSCALE)
```

Comment:
also need to note about expected semantic mask format (in some cases, it can be encoded by colors)